### PR TITLE
Fix UndefVarError

### PR DESCRIPTION
**Pull request by michel2323**
* Fixed UndefVarError introduced in https://github.com/JuliaGPU/KernelAbstractions.jl/pull/236

Moving from `v0.6.0` to `v0.6.1` triggered a few undefined variable errors. This PR fixes those.

### CODE REVIEW

**Comment by vchuravy** [comment]
Can you also add a test-case?

**Comment by michel2323** [comment]
@vchuravy Added an `:` operator test, but not sure in the right place.

**Comment by michel2323** [comment]
Okay, works with the CPU but not with the CUDAKernels backend. I guess `:` should not be used this way.

File: test/colon.jl [hunk 1]
```diff
@@ -0,0 +1,15 @@
+using KernelAbstractions, Test
+
+@kernel function copy_colon_kernel!(A, @Const(B))
+    I = @index(Global)
+    @inbounds A[I] = sum(B[:,I])
```
Comment (vchuravy) [inline test/colon.jl:5]:
```suggestion
    @inbounds A[I] = sum(view(B, :,I))
```
maybe?

Comment (michel2323) [reply test/colon.jl:5]: That solves our problem in `ExaPF`, but then doesn't trigger the `unsafe_getindex` overdub in https://github.com/JuliaGPU/KernelAbstractions.jl/blob/51a59dc8ff2427b53db93a2e50efb4a972e370f5/src/compiler.jl#L94 .

Comment (michel2323) [reply test/colon.jl:5]: https://github.com/exanauts/ExaPF.jl/blob/34f3a7103851f04c61b578808880ad64aeba739d/src/LinearSolvers/preconditioners.jl#L277  this is another statement that breaks for the same reason but without `:`.

**Comment by michel2323** [comment]
@vchuravy I couldn't get an MWE where `_unsafe_getindex` is called other than with the colon, although we get it triggered in ExaPF. However, I could not get why. So here is a test that directly calls `_unsafe_getindex`.

**Comment by michel2323** [comment]
Bad bot.

**Comment by michel2323** [comment]
@vchuravy I guess you have to decide whether `_unsafe_getindex` should be supported. As of now [this code ](https://github.com/JuliaGPU/KernelAbstractions.jl/blob/51a59dc8ff2427b53db93a2e50efb4a972e370f5/src/compiler.jl#L94-L100) only works on the CPU backend. @jakebolewski What's your take?

**Comment by jakebolewski** [comment]
I would revert that PR, I could never get it to work correctly (which is why I couldnt write the test and shelved it a while back) and we are going in a different direction with some of our project's kernels so I don't think it is useful in the near term.

**Comment by vchuravy** [comment]
Fixed by  #246

**Pull request closed without merging**